test/profiler/test_profiler_tree.py

```diff
@@ -191,6 +191,14 @@ def fmt_name(name: str) -> str:
                 name,
             )
 
+        # HACK: this patches around the fact that PyBind11 improperly sets the
+        # __qualname__ attribute on functions and methods; see
+        # https://github.com/pybind/pybind11/issues/5774.  This should be removed if
+        # that issue is fixed.
+        name = "PyCapsule".join(
+            re.split(r"pybind11_detail_function_record_v1[^ ]+", name, maxsplit=1)
```

Comment:
Maybe make this `pybind11_detail_function_record_v` (excluding the `1`), just in case (however unlikely)?